src/mutation/updateItem.ts

```diff
@@ -29,18 +30,39 @@ const buildConditionExpressions = (
       const value = currentExpression.value;
 
       if (!comparator) continue;
-      const operator = keyOperatorLookup[comparator];
+      const operator = keyOperatorLookup(comparator);
+      const expressionName = `#key_${key}`;
+      attrNames[expressionName] = key;
+
       if (operator === 'BETWEEN') {
-        expression += `${key} ${operator} :val${i}_1 AND :val${i}_2`;
-        attrValues[`:val${i}_1`] = { N: value[0].toString() };
-        attrValues[`:val${i}_2`] = { N: value[1].toString() };
+        expression += `${expressionName} ${operator} :val${i}_1 AND :val${i}_2`;
+        attrValues[`:val${i}_1`] = value[0];
+        attrValues[`:val${i}_2`] = value[1];
       } else {
-        expression += `${key} ${operator} :val${i}`;
-        attrValues[`:val${i}`] = { N: value.toString() };
+        expression += `${expressionName} ${operator} :val${i}`;
+        attrValues[`:val${i}`] = value;
       }
     }
   }
-  return { expression, attrValues };
+  return { expression, attrValues, attrNames };
+};
+
+const buildUpdateExpressions = (item: object) => {
+  const expressions = [];
+  const expressionValues = {};
+  const expressionNames = {};
+
+  Object.keys(item)?.forEach(key => {
+    expressions.push(`#key_${key} = :val_${key}`);
+    expressionNames[`#key_${key}`] = key;
+    expressionValues[`:val_${key}`] = item[key];
+  });
+
+  return {
```

Comment:
I didn't give attention to the `ConditionExpressionReturn` type that existed. Please use that for the return value for parity with [condition expr](https://github.com/hitz-group/dynamo-helper/pull/132/files#diff-6939908fe56505e862a3e96f2bd0f0a6be0e8cb0ebc829060a5b4d244ace0a8cR103)

Reply:
updated